### PR TITLE
feat: added wishlist functionality

### PR DESCRIPTION
**Pull request by amarphule**
Added Wishlist functionality:
1. User can add a product to wishlist.
2. Wishlist count we'll see on the navbar.
3. User can remove added product item from wishlist.

### CODE REVIEW

File: src/Pages/Wishlist/WishList.js [hunk 1]
```diff
@@ -1,71 +1,28 @@
...
-import "./WishList.css"
+import React from "react";
+import { useWishList } from "../../Contexts/WishListContext";
+import Product from "../../Components/Product";
+
+import "./WishList.css";
 
 const WishList = () => {
+  const { wishlist } = useWishList();
+  console.log(wishlist);
```
Comment (prathmeshjagtap) [inline src/Pages/Wishlist/WishList.js:9]: remove console.log

**Review by prathmeshjagtap — APPROVED** [review]

File: src/Components/Product.js [hunk 2]
```diff
@@ -13,12 +14,36 @@ const Product = ({ product }) => {
   const removeFromCartHandler = () => {
     setIsInCart(false);
   };
+
+  const { wishlist, setWishlist } = useWishList();
+
+  const addWishlistHandler = (e) => {
+    e.preventDefault();
```
Comment (horsemaker) [inline src/Components/Product.js:21]: Why `e.preventDefault()` on a button click?

**Review by horsemaker — APPROVED** [review]
Looks good! 👌

**Review by t007rushi — APPROVED** [review]
Everything Looks fine to me

File: src/Components/Product.js (hunk 2 above)
Comment (amarphule) [reply src/Components/Product.js:21]: Thank you, made changes

**Pull request merged**